### PR TITLE
[WIP][Society] Adding benchmarking to the v2.

### PR DESCRIPTION
**Pull request by arturgontijo**
This PRs tracks https://github.com/paritytech/substrate/pull/11324 and adds `benchmarking.rs` to it.

I had to touch on some other files in order to make it compile.

### CODE REVIEW

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -1313,14 +1313,14 @@ impl pallet_recovery::Config for Runtime {
 }
 
 parameter_types! {
-	pub const CandidateDeposit: Balance = 10 * DOLLARS;
-	pub const WrongSideDeduction: Balance = 2 * DOLLARS;
-	pub const MaxStrikes: u32 = 10;
-	pub const RotationPeriod: BlockNumber = 80 * HOURS;
+	pub const GraceStrikes: u32 = 10;
+	pub const SocietyVotingPeriod: BlockNumber = 80 * HOURS;
```
Comment (kianenigma) [inline bin/node/runtime/src/lib.rs:1317]: Easier to express this with `DAYS`?

File: frame/society/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+}
+
+fn increment_round<T: Config<I>, I: 'static>() {
+	let mut round_count = RoundCount::<T, I>::get();
+	round_count.saturating_inc();
+	RoundCount::<T, I>::put(round_count);
+}
+
+benchmarks_instance_pallet! {
+
```
Comment (kianenigma) [inline frame/society/src/benchmarking.rs:77]:
```suggestion
```

File: frame/society/src/benchmarking.rs [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+	bid {
+		let founder: T::AccountId = account("founder", 0, 0);
+		setup_society::<T, I>(founder)?;
+
+		let caller: T::AccountId = whitelisted_caller();
+		let deposit: BalanceOf<T, I> = T::Currency::minimum_balance();
+		T::Currency::make_free_balance_be(&caller, BalanceOf::<T, I>::max_value());
+	}: _(RawOrigin::Signed(caller.clone()), 10u32.into())
```
Comment (kianenigma) [inline frame/society/src/benchmarking.rs:85]: You should bit with a multiple of `minimum_balance`. This value is smaller than the existential deposit in all real chains and might have unforeseen consequences.

File: frame/society/src/weights.rs [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+			.saturating_add(T::DbWeight::get().reads(2 as Weight))
+			.saturating_add(T::DbWeight::get().writes(7 as Weight))
+	}
+	// Storage: Society Founder (r:1 w:1)
+	// Storage: Society MemberCount (r:1 w:1)
+	// Storage: Society Head (r:0 w:1)
+	// Storage: Society MemberByIndex (r:0 w:1)
+	// Storage: Society Rules (r:0 w:1)
+	// Storage: Society Members (r:0 w:1)
+	fn dissolve() -> Weight {
```
Comment (kianenigma) [inline frame/society/src/weights.rs:154]: a bit too good to be true. The dissolve benchmark should assume "worse case", which means there should be maximum number of votes and other data structures also around to clean.

**Review by kianenigma — COMMENTED** [review]
Looks mostly good, should avoid hard-coding numbers as balance.

**Comment by arturgontijo** [comment]
This one is ready for review.

File: bin/node/runtime/src/lib.rs (hunk 1 above)
Comment (gavofyork) [reply bin/node/runtime/src/lib.rs:1317]: I explicitly made it this way to avoid giving one global region preference in daylight times.

**Pull request merged**